src/pages/cookbooks/[cookbookId]/index.tsx

```diff
@@ -3,24 +3,39 @@ import { useRouter } from 'next/router';
 import { ROUTES } from '../../../constants/constants';
 import HttpService from '../../../utils/HttpService';
 import { useStore } from '../../../store/store';
-
-const init = async (cookbookId?: string | string[]) => {
-  if (cookbookId != null) {
-    const cookbook = await HttpService.getById(ROUTES.COOKBOOKS, cookbookId);
-    const guides = await HttpService.get(ROUTES.GUIDES(cookbook._id));
-    useStore.setState({ cookbook, guides });
-  }
-};
+import { useState, useEffect } from 'react';
+import { TwitchSidebar } from '../../../components/Sidebar/TwitchSidebar/TwitchSidebar';
 
 export const CookbookHome: React.FC = () => {
   const router = useRouter();
   const { cookbookId } = router.query;
+  const [cookbook, setCookbook] = useState(useStore.getState().cookbook);
+  const [streams, setstreams] = useState(useStore.getState().cookbook?.streams);
+
+  useEffect(() => {
+    const init = async (cookbookId?: string | string[]) => {
+      if (cookbookId != null) {
+        const cookbook = useStore
+          .getState()
+          .cookbooks.find(cookbook => cookbook._id === cookbookId);
+
+        if (cookbook) {
+          setCookbook(cookbook);
+          const guides = await HttpService.get(ROUTES.GUIDES(cookbook._id));
+          useStore.setState({ cookbook, guides });
+          setstreams(cookbook.streams);
```

Comment:
If I get this right, this method locally gets the cookbook from the store, sets it and its streams in the store and then uses the store to pass the streams to the sidebar. Why not just grab it and store it locally?

Reply:
not sure i follow here. i'm not storing streams in the store, it's just a useState hook for setting streams in this component, so thats a local store, no?

as this is the cookbook/id page we should fetch the cookbook from the store based on the route ID, and then we locally store the streams with that cookbook. should i do it differently?

Reply:
alright np!